src/deepsparse/pipeline.py

```diff
@@ -222,39 +258,23 @@ def __call__(self, *args, monitoring: bool = False, **kwargs) -> BaseModel:
                 f"Outputs of {self.__class__} must be instances of "
                 f"{self.output_schema} found output of type {type(pipeline_outputs)}"
             )
+        if self.logger:
+            self.logger.log(
+                identifier=f"{self._alias}.pipeline_outputs",
+                value=pipeline_outputs,
+                category=self.logger.metric_categories.DATA,
+            )
         timer.stop(InferencePhases.POST_PROCESS)
         timer.stop(InferencePhases.TOTAL_INFERENCE)
 
-        if not monitoring:
-            return pipeline_outputs
-
-        else:
-            inference_timing = InferenceTimingSchema(**timer.build())
-            return pipeline_outputs, pipeline_inputs, engine_inputs, inference_timing
+        if self.logger:
+            self.logger.log(
+                identifier=f"{self._alias}.inference_timing",
+                value=timer.build(),
```

Comment:
Instead of passing the result of `timer.build(): dict` to `InferenceTimingSchema` (like we did before), we can pass the dict to the logger and then further alter it inside the logger, not the pipeline.

Reply:
we should be getting rid of inference timing schema now - this is because we want to be logging the timings in real time as they happen in case there is an error

Reply:
Yeah, makes sense.